std/valgrind/index.zig

```diff
@@ -0,0 +1,367 @@
+        0, 0, 0);
+}
+
+
+// Create a memory pool.
+pub const MempoolFlags = extern enum {
+    AutoFree = 1,
+    MetaPool = 2,
+};
+pub fn createMempool(pool: [*]u8, rzB: usize, is_zeroed: bool, flags: usize) void {
+    doClientRequestStmt(ClientRequest.CreateMempool,
+        @ptrToInt(pool), @ptrToInt(rzB), @boolToInt(is_zeroed), flags,
+        0);
+}
+
+// Destroy a memory pool.
+pub fn destroyMempool(pool: [*]u8) void {
+    doClientRequestStmt(ClientRequest.DestroyMempool,
+        pool,
+        0, 0, 0, 0);
+}
+
+
+// Associate a piece of memory with a memory pool.
+pub fn mempoolAlloc(pool: [*]u8, mem: []u8) void {
+    doClientRequestStmt(ClientRequest.MempoolAlloc,
+        @ptrToInt(pool), @ptrToInt(mem.ptr), mem.len,
+        0, 0);
+}
+
+// Disassociate a piece of memory from a memory pool.
+pub fn mempoolFree(pool: [*]u8, addr: [*]u8) void {
+    doClientRequestStmt(ClientRequest.MempoolFree,
+        @ptrToInt(pool), @ptrToInt(addr),
+        0, 0, 0);
+}
+
+// Disassociate any pieces outside a particular range.
+pub fn mempoolTrim(pool: [*]u8, mem: []u8) void {
+    doClientRequestStmt(ClientRequest.MempoolTrim,
+        @ptrToInt(pool), @ptrToInt(mem.ptr), mem.len,
+        0, 0);
+}
+
+// Resize and/or move a piece associated with a memory pool.
+pub fn moveMempool(poolA: [*]u8, poolB: [*]u8) void {
+    doClientRequestStmt(ClientRequest.MoveMempool,
+        @ptrToInt(poolA), @ptrToInt(poolB),
+        0, 0, 0);
+}
+
+// Resize and/or move a piece associated with a memory pool.
+pub fn mempoolChange(pool: [*]u8, addrA: [*]u8, mem: []u8) void {
+    doClientRequestStmt(ClientRequest.MempoolChange,
+        @ptrToInt(pool), @ptrToInt(addrA), @ptrToInt(mem.ptr), mem.len,
+        0);
+}
+
+// Return 1 if a mempool exists, else 0.
+pub fn mempoolExists(pool: [*]u8) bool {
+    return doClientRequestExpr(0,
+        ClientRequest.MempoolExists,
+        pool,
+        0, 0, 0, 0) != 0;
+}
+
+
+// Mark a piece of memory as being a stack. Returns a stack id.
+// start is the lowest addressable stack byte, end is the highest
+// addressable stack byte.
+pub fn stackRegister(stack: []u8) usize {
+    return doClientRequestExpr(0,
+        ClientRequest.StackRegister,
+        @ptrToInt(stack.ptr), @ptrToInt(stack.ptr) + stack.len,
+        0, 0, 0);
+}
+
+// Unmark the piece of memory associated with a stack id as being a stack.
+pub fn stackDeregister(id: usize) void {
+    doClientRequestStmt(ClientRequest.StackDeregister,
+        id,
+        0, 0, 0, 0);
+}
+
+// Change the start and end address of the stack id.
+// start is the new lowest addressable stack byte, end is the new highest
+// addressable stack byte.
+pub fn stackChange(id: usize, newstack: []u8) void {
+    doClientRequestStmt(ClientRequest.StackChange,
+        id, @ptrToInt(newstack.ptr), @ptrToInt(newstack.ptr) + newstack.len,
+        0, 0);
+}
+
+
+// Load PDB debug info for Wine PE image_map.
+// pub fn loadPdbDebuginfo(fd, ptr, total_size, delta) void {
```

Comment:
I have no idea how this function is meant to work or what the args are; it is underdocumented.